client/core/core.go

```diff
@@ -208,6 +216,82 @@ func (c *Core) handleReconnect(uri string) {
 	log.Infof("DEX at %s has reconnected", uri)
 }
 
+// handleOrderBookMsg is called when an orderbook request is received.
+func (c *Core) handleOrderBookMsg(dc *dexConnection, msg *msgjson.Message) error {
+	resp, err := msg.Response()
+	if err != nil {
+		return err
+	}
+
+	var snapshot msgjson.OrderBook
+	err = json.Unmarshal(resp.Result, &snapshot)
+	if err != nil {
+		return fmt.Errorf("order book unmarshal error: %v", err)
+	}
+
+	ob := order.NewOrderBook()
+	err = ob.Sync(&snapshot)
+	if err != nil {
+		return err
+	}
+
+	dc.booksMtx.Lock()
+	dc.books[snapshot.MarketID] = ob
+	dc.booksMtx.Unlock()
+
+	return nil
+}
+
+// handleBookOrderMsg is called when a book_order notification is received.
+func (c *Core) handleBookOrderMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.BookOrderNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("book order note unmarshal error: %v", err)
+	}
+
+	dc.booksMtx.Lock()
+	defer dc.booksMtx.Unlock()
+
+	ob, ok := dc.books[note.MarketID]
+	if !ok {
+		return fmt.Errorf("no order book found with market id '%v'",
+			note.MarketID)
+	}
+
+	err = ob.Book(&note)
```

Comment:
Is `booksMtx` also intended to guard the `ob` states as well, or is it intended to protect just the map `dc.books`?  `OrderBook` has it's internal synchronization and `booksMtx` implies it's just for the map.  If it's just the map, might as well `dc.booksMtx.Unlock()` immediately after `ob, ok := dc.books[note.MarketID]`.